osu.Game.Tests/Visual/Online/TestSceneOnlineViewContainer.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using NUnit.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Online;
+using osu.Game.Online.API;
+using osuTK.Graphics;
+
+namespace osu.Game.Tests.Visual.Online
+{
+    [TestFixture]
+    public class TestSceneOnlineViewContainer : OsuTestScene
+    {
+        private readonly OnlineViewContainer onlineView;
+
+        public TestSceneOnlineViewContainer()
+        {
+            Child = new Container
+            {
+                RelativeSizeAxes = Axes.Both,
+                Child = onlineView = new OnlineViewContainer(@"Please sign in to view dummy test content")
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Children = new Drawable[]
+                    {
+                        new Box
+                        {
+                            RelativeSizeAxes = Axes.Both,
+                            Colour = Color4.Blue.Opacity(0.8f),
+                        },
+                        new OsuSpriteText
+                        {
+                            Text = "dummy online content",
+                            RelativeSizeAxes = Axes.Both,
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                        }
+                    }
+                }
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            AddStep("set status to offline", () => ((DummyAPIAccess)API).State = APIState.Offline);
+
+            AddAssert("children are hidden", () => !onlineView.Children.First().Parent.IsPresent);
```

Comment:
I advocate you expose `View` instead of playing with the sub-tree to retrieve it.

also, checking is present is not the best way to determine if a drawable is hidden, it may have a `override IsPresent => true` or custom values because it's a manager or something, check `Alpha` directly instead.

Reply:
Actually, since View is a normal container, checking if drawable is visible through IsPresent should be safe

Reply:
Well, using `IsPresent` is technically not safe because you're not checking alpha, but you're checking a property that has alpha in it's calculation, I find it more arbitrary.

Reply:
Searching for IsPresent in Tests yields results where that property is being used to check if a drawable is displayed on screen. So it should be fine to use it to check for drawable visibility
![Capture](https://user-images.githubusercontent.com/20256717/73134622-10f49780-4039-11ea-92cd-c3da5898ebc2.PNG)